grobid-core/src/test/java/org/grobid/core/sax/PDFALTOOutlineSaxHandlerTest.java

```diff
@@ -0,0 +1,74 @@
+package org.grobid.core.sax;
+
+import org.grobid.core.document.Document;
+import org.grobid.core.document.DocumentSource;
+import org.grobid.core.document.DocumentNode;
+import org.junit.Before;
+import org.junit.Test;
+
+import javax.xml.parsers.SAXParser;
+import javax.xml.parsers.SAXParserFactory;
+
+import java.io.InputStream;
+
+import static org.easymock.EasyMock.createMock;
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.CoreMatchers.nullValue;
+import static org.hamcrest.collection.IsCollectionWithSize.hasSize;
+import static org.junit.Assert.assertThat;
+import static org.junit.Assert.assertTrue;
+
+public class PDFALTOOutlineSaxHandlerTest {
+    SAXParserFactory spf = SAXParserFactory.newInstance();
+
+    PDFALTOOutlineSaxHandler target;
+    DocumentSource mockDocumentSource;
+    Document document;
+
+    @Before
+    public void setUp() throws Exception {
+
+        mockDocumentSource = createMock(DocumentSource.class);
+
+        document = Document.createFromText("");
+        target = new PDFALTOOutlineSaxHandler(document);
+    }
+
+    @Test
+    public void testParsing_pdf2XMLOutline_ShouldWork() throws Exception {
+        InputStream is = this.getClass().getResourceAsStream("pdfalto.xml_outline.xml");
```

Comment:
I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [UI: Usage of GetResource may be unsafe if class is extended (UI_INHERITANCE_UNSAFE_GETRESOURCE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#ui-usage-of-getresource-may-be-unsafe-if-class-is-extended-ui-inheritance-unsafe-getresource).
Calling this.getClass().getResource(...) could give results other than expected if this class is extended by a class in another package.

Reply:
Hi, thanks for your feedback. This is the code of a unit test, it's running only after compilation and it's not supposed to be extended. Do you have any suggestion on how to change it?

Reply:
Since this class is not extended by a class, it should be safe. Thanks a lot!

Reply:
thank you for your quick reply. Since this is a test class, this problem is inapplicable